python/feathub/feature_tables/sources/hive_source.py

```diff
@@ -0,0 +1,118 @@
+#  Copyright 2022 The FeatHub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import hashlib
+import os.path
+import random
+import string
+from typing import Optional, Dict, List
+
+from feathub.common.utils import is_local_file_or_dir
+from feathub.feature_tables.feature_table import FeatureTable
+from feathub.table.schema import Schema
+
+
+def get_hive_catalog_identifier(hive_catalog_conf_dir: str) -> str:
+    """
+    Return an identifier for the hive catalog configuration. Configurations
+    with the same identifier refer to the same Hive catalog.
+    """
+    if is_local_file_or_dir(hive_catalog_conf_dir):
+        hive_catalog_conf_file = os.path.join(hive_catalog_conf_dir, "hive-site.xml")
+        hash_md5 = hashlib.md5()
+        with open(hive_catalog_conf_file, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+        return hash_md5.hexdigest()
+    else:
+        return "".join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(32)
+        )
+
+
+class HiveSource(FeatureTable):
+    """
+    A source that reads data from Hive.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        database: str,
+        table: str,
+        hive_catalog_conf_dir: str,
+        schema: Schema,
+        keys: Optional[List[str]] = None,
+        timestamp_field: Optional[str] = None,
+        timestamp_format: str = "epoch",
+        extra_config: Optional[Dict[str, str]] = None,
```

Comment:
In the future we might want to introduce FeatHub-recognized configs for these sources and sinks and translate these configs into processor-specific configs. And we don't want FeatHub-recognized configs to conflict with processor-specific configs in the same map.

How about renaming it `processor_specific_props`?